core/src/main/java/org/sonarsource/sonarlint/core/ConnectionService.java

```diff
@@ -168,33 +156,32 @@ private void updateConnection(AbstractConnectionConfiguration connectionConfigur
     }
   }
 
-  public ValidateConnectionResponse validateConnection(ValidateConnectionParams params, CancelChecker cancelToken) {
-    var helper = buildServerApiHelper(params.getTransientConnection());
+  public ValidateConnectionResponse validateConnection(Either<TransientSonarQubeConnectionDto, TransientSonarCloudConnectionDto> transientConnection, CancelChecker cancelToken) {
+    var helper = buildServerApiHelper(transientConnection);
     var connectionValidator = new ConnectionValidator(helper);
     var r = connectionValidator.validateConnection(cancelToken);
     return new ValidateConnectionResponse(r.success(), r.message());
   }
 
-  public CheckSmartNotificationsSupportedResponse checkSmartNotificationsSupported(CheckSmartNotificationsSupportedParams params, CancelChecker cancelToken) {
-    var helper = buildServerApiHelper(params.getTransientConnection());
+  public boolean checkSmartNotificationsSupported(Either<TransientSonarQubeConnectionDto, TransientSonarCloudConnectionDto> transientConnection) {
+    var helper = buildServerApiHelper(transientConnection);
     var developersApi = new ServerApi(helper).developers();
-    return new CheckSmartNotificationsSupportedResponse(developersApi.isSupported());
+    return developersApi.isSupported();
   }
 
-  public ListUserOrganizationsResponse listUserOrganizations(ListUserOrganizationsParams params, CancelChecker cancelToken) {
-    var helper = buildSonarCloudNoOrgApiHelper(params.getCredentials());
+  public List<OrganizationDto> listUserOrganizations(Either<TokenDto, UsernamePasswordDto> credentials) {
+    var helper = buildSonarCloudNoOrgApiHelper(credentials);
     var serverOrganizations = new OrganizationApi(helper).listUserOrganizations(new ProgressMonitor(null));
-    return new ListUserOrganizationsResponse(
-      serverOrganizations.stream().map(o -> new OrganizationDto(o.getKey(), o.getName(), o.getDescription())).collect(Collectors.toList()));
+    return serverOrganizations.stream().map(o -> new OrganizationDto(o.getKey(), o.getName(), o.getDescription())).collect(Collectors.toList());
   }
 
-  public GetOrganizationResponse getOrganization(GetOrganizationParams params, CancelChecker cancelToken) {
-    var helper = buildSonarCloudNoOrgApiHelper(params.getCredentials());
-    var serverOrganization = new OrganizationApi(helper).getOrganization(params.getOrganizationKey(), new ProgressMonitor(null));
-    return new GetOrganizationResponse(serverOrganization.map(o -> new OrganizationDto(o.getKey(), o.getName(), o.getDescription())).orElse(null));
+  @CheckForNull
+  public OrganizationDto getOrganization(Either<TokenDto, UsernamePasswordDto> credentials, String organizationKey) {
```

Comment:
I wonder if we shouldn't do the same for DTOs too, as they are normally specific to RPC

Reply:
That would be the next step if we want complete isolation between layers, but then we will have more boilerplate code. I suggest we keep using Dtos when needed, and we can adjust later.